packages/engine/src/datastore/arrow/ipc.rs

```diff
@@ -26,652 +26,22 @@
-        DataType::Time32(TimeUnit::Second) => Arc::new(Time32SecondArray::from(data)),
-        DataType::Time32(TimeUnit::Millisecond) => Arc::new(Time32MillisecondArray::from(data)),
-        DataType::Time64(TimeUnit::Microsecond) => Arc::new(Time64MicrosecondArray::from(data)),
-        DataType::Time64(TimeUnit::Nanosecond) => Arc::new(Time64NanosecondArray::from(data)),
-        DataType::Timestamp(TimeUnit::Second, _) => Arc::new(TimestampSecondArray::from(data)),
-        DataType::Timestamp(TimeUnit::Millisecond, _) => {
-            Arc::new(TimestampMillisecondArray::from(data))
-        }
-        DataType::Timestamp(TimeUnit::Microsecond, _) => {
-            Arc::new(TimestampMicrosecondArray::from(data))
-        }
-        DataType::Timestamp(TimeUnit::Nanosecond, _) => {
-            Arc::new(TimestampNanosecondArray::from(data))
-        }
-        DataType::Interval(IntervalUnit::YearMonth) => Arc::new(IntervalYearMonthArray::from(data)),
-        DataType::Interval(IntervalUnit::DayTime) => Arc::new(IntervalDayTimeArray::from(data)),
-        DataType::Duration(TimeUnit::Second) => Arc::new(DurationSecondArray::from(data)),
-        DataType::Duration(TimeUnit::Millisecond) => Arc::new(DurationMillisecondArray::from(data)),
-        DataType::Duration(TimeUnit::Microsecond) => Arc::new(DurationMicrosecondArray::from(data)),
-        DataType::Duration(TimeUnit::Nanosecond) => Arc::new(DurationNanosecondArray::from(data)),
-        DataType::Binary => Arc::new(BinaryArray::from(data)),
-        DataType::FixedSizeBinary(_) => Arc::new(FixedSizeBinaryArray::from(data)),
-        DataType::Utf8 => Arc::new(StringArray::from(data)),
-        DataType::List(_) => Arc::new(ListArray::from(data)),
-        DataType::Struct(_) => Arc::new(StructArray::from(data)),
-        DataType::Union(_) => Arc::new(UnionArray::from(data)),
-        DataType::FixedSizeList(..) => Arc::new(FixedSizeListArray::from(data)),
-        DataType::Dictionary(ref key_type, _) => match key_type.as_ref() {
-            DataType::Int8 => Arc::new(DictionaryArray::<Int8Type>::from(data)),
-            DataType::Int16 => Arc::new(DictionaryArray::<Int16Type>::from(data)),
-            DataType::Int32 => Arc::new(DictionaryArray::<Int32Type>::from(data)),
-            DataType::Int64 => Arc::new(DictionaryArray::<Int64Type>::from(data)),
-            DataType::UInt8 => Arc::new(DictionaryArray::<UInt8Type>::from(data)),
-            DataType::UInt16 => Arc::new(DictionaryArray::<UInt16Type>::from(data)),
-            DataType::UInt32 => Arc::new(DictionaryArray::<UInt32Type>::from(data)),
-            DataType::UInt64 => Arc::new(DictionaryArray::<UInt64Type>::from(data)),
-            dt => panic!("Unexpected dictionary key type {:?}", dt),
-        },
-        DataType::Null => Arc::new(NullArray::from(data)),
-        dt => panic!("Unexpected data type {:?}", dt),
-    }
-}
-
-// COPY: ::ipc::writer.rs
-// MOD: added continuation bytes for Python read
```

Comment:
So far we don't need the changes. Also, the commit doing this changes passed the CI just fine.
When adjusting arrow for Python, this may need some adjustments again, but I want to focus on JS for now.

Reply:
Sounds good, as long as we keep track of them

Reply:
I updated the description of the corresponding [Asana task](https://app.asana.com/0/0/1201829883247749/f) (_internal_).